docs/DeveloperGuide.md

```diff
@@ -1,5 +1,166 @@
+WritingList: Represent the objects which are particular lists of Writings to be used in the application.
+Writings: Represent the objects of the writings, created from user’s input and stored in a database as text. 
+This Writings class is also the parent of 2 subclasses which are Poem and Essay.
+Poem: Represents the Writings objects which have type is Poem.
+Essay: Represents the Writings objects which have type is Essay.
+User: Represents the Users registered to the System
+![UML Class diagram for WritingList family](graphics/diagrams/classDiagram_WritingList.png)
+<p align = "center"><i><b>Figure 2: WritingList family UML diagram</b></i></p>
+
+The above class diagram describes the overall architecture of Writings class functionalities and associations within 
+the scope of related classes. By checking “start”, “type” command with checkStartCommand() then checkTypeCommand() 
+methods on that sequence respectively, the user should be able to access the process of creating and saving new writings 
+into the database. During this process, the user has the ability of choosing their preferred type of writings(which are 
+either poem or essay at this stage)
+
+![UML WritingList family sequence diagram](graphics/diagrams/UML_Writing_Sequence_diagram.png)
+<p align = "center"><i><b>Figure 3: General interactions between member classes when generating a new writing</b></i></p>
+
+### Filter words class family
+![UML Filter word class diagram](graphics/diagrams/classDiagram_FilterWords.png)
+<p align = "center"><i><b>Figure 4: Filter word UML Class Diagram</b></i></p>
+
+The above class diagram describes the overall architecture of the filter words functionality. `FilterExecutor` class has 
+the static void method `executeFilterCommand` that will be called first when the user enters a filter command. 
+In the `executeFilterCommand` method, it will make use of the enumeration `FilterType` to get the filter type (by `WORD_TYPE`, 
+`STARTING STRING` or `INCLUDING_STRING`). After that, the method will use the `FilterCommandSlicer` static methods `isNewFilter` 
+to determine whether the user wants to continue on the last filtered list or start a new filter on an entire word bank. 
+`executeFilterCommand` will also check whether the user has entered a print limit using `FilterCommandSlicer`'s method 
+`getWordPrintLimitFromFilterCommand`. Subsequently, depending on the filter mode, `getTargetedWordTypes` or `getTargetedStringTags` will be called 
+and the returned array of strings will be passed to `WordsFilter`’s static methods `filterByType`, `filterByStartingString` and `filterByIncludedString`.
+Lastly, `filterCommandExecutor` will call the method `printFilterList`.
+
+The following sequence diagram shows how the components interact with each other for the scenario where the user issues 
+the command `filter -continue by\start limit\10 -cs -cg.`
+
+![UML Filter word sequence diagram](graphics/diagrams/Sequence_FilterWords.png)
```

Comment:
Is it necessary to include " << class >> " for all classes? Maybe you could also include a return arrow to return "null" for any methods that do not return anything.
<img width="891" alt="Screenshot 2020-10-28 at 11 16 01 AM" src="https://user-images.githubusercontent.com/49362047/97397928-57508d00-1910-11eb-8735-19b7856d62f4.png">